### PR TITLE
Add new years resolution badge

### PR DESCRIPTION
**Pull request by ErikSchierboom**
This query will add the badge to around 2300 users (once https://github.com/exercism/website/pull/2504/files is merged):

```ruby
# 2303 users will get this badge
new_years_resolution_user_ids = 
    Arel.sql(
        Solution.completed.
            where('DAYOFYEAR(solutions.created_at) = 1').
            select(:user_id).
            distinct.
            to_sql)
User.where("id IN (#{new_years_resolution_user_ids})").find_each do |user|
    AwardBadgeJob.perform_later(user, :new_years_resolution, send_email: false)
end
```

### CODE REVIEW

File: app/images/icons/new-years-resolution.svg [hunk 1]
```diff
@@ -0,0 +1,10 @@
+<svg width="30" height="30" viewBox="0 0 30 30" fill="none" xmlns="http://www.w3.org/2000/svg">
```
Comment (ErikSchierboom) [inline app/images/icons/new-years-resolution.svg]: @taiyab Another icon for you :)

File: app/commands/solution/create.rb [hunk 2]
```diff
@@ -9,6 +9,7 @@ def call
       begin
         solution_class.create!(user: user, exercise: exercise).tap do |solution|
           record_activity!(solution)
+          AwardBadgeJob.perform_later(user, :new_years_resolution)
```
Comment (ErikSchierboom) [inline app/commands/solution/create.rb:12]: I've currently awarded the badge when a solution is created, not when it was completed. Let me know if that is okay.

File: app/models/badges/new_years_resolution_badge.rb [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+  class NewYearsResolutionBadge < Badge
+    seed "New Years Resolution",
+      :rare,
+      'new-years-resolution',
+      'Awarded for submitting a solution on January 1st'
+
+    def award_to?(user)
+      user.solutions.
+        where('DAYOFYEAR(solutions.created_at) = 1').
+        exists?
```
Comment (ErikSchierboom) [inline app/models/badges/new_years_resolution_badge.rb:11]: As we don't yet store the users local time (I think), this does mean that the current implementation will be incorrect for users in non-UTC timezones. I'm not yet sure what we should be doing about that.

**Comment by iHiD** [comment]
```
<svg width="700" height="700" viewBox="0 0 700 700" fill="none" xmlns="http://www.w3.org/2000/svg">
<path d="M189.163 0.0816425C181.345 0.941678 174.825 8.37121 174.992 16.2335V127.544C174.876 135.944 182.494 143.673 190.9 143.673C199.306 143.673 206.929 135.944 206.807 127.544V16.2335C207.001 7.21142 198.129 -0.905565 189.157 0.0816425H189.163ZM507.342 0.0816425C499.525 0.941678 493.005 8.37121 493.171 16.2335V127.544C493.055 135.944 500.673 143.673 509.079 143.673C517.485 143.673 525.109 135.944 524.986 127.544V16.2335C525.181 7.21142 516.308 -0.905565 507.336 0.0816425H507.342ZM87.5021 55.9832C39.3008 55.9832 0 95.2613 0 143.44V596.632C0 644.811 39.3008 684.089 87.5021 684.089H612.498C660.699 684.089 700 644.811 700 596.632V143.44C700 95.2613 660.699 55.9832 612.498 55.9832H572.725C564.319 55.8667 556.591 63.4848 556.591 71.885C556.591 80.2857 564.32 87.9034 572.725 87.7869H612.498C643.62 87.7869 668.178 112.334 668.178 143.44V596.632C668.178 627.737 643.62 652.285 612.498 652.285H87.5021C56.3802 652.285 31.8222 627.738 31.8222 596.632V230.897H620.454C628.86 231.013 636.589 223.395 636.589 214.995C636.589 206.594 628.859 198.976 620.454 199.093H31.8222V143.44C31.8222 112.335 56.3802 87.7869 87.5021 87.7869H127.275C135.681 87.9034 143.409 80.2853 143.409 71.885C143.409 63.4844 135.68 55.8667 127.275 55.9832H87.5021ZM253.055 55.9832C244.726 56.3716 237.497 64.3006 237.89 72.6294C238.279 80.9578 246.219 88.1761 254.548 87.7826H445.455C453.861 87.8992 461.59 80.281 461.59 71.8808C461.59 63.4801 453.861 55.8624 445.455 55.9789H253.056L253.055 55.9832Z" fill="black"/>
<path d="M309.4 536C302.307 536 298.76 532.64 298.76 525.92C298.76 519.573 302.307 516.4 309.4 516.4H345.8V365.48L317.24 384.52C314.253 386.387 311.453 386.947 308.84 386.2C306.413 385.267 304.547 383.68 303.24 381.44C301.933 379.2 301.467 376.773 301.84 374.16C302.4 371.547 304.173 369.213 307.16 367.16L345.8 341.68C348.227 340.187 350.56 338.973 352.8 338.04C355.04 337.107 357.28 336.64 359.52 336.64C362.133 336.64 364.28 337.48 365.96 339.16C367.827 340.653 368.76 343.08 368.76 346.44V516.4H402.36C409.453 516.4 413 519.573 413 525.92C413 532.64 409.453 536 402.36 536H309.4Z" fill="black"/>
</svg>
```

**Pull request merged**